### PR TITLE
Updated the context object in all services.

### PR DESCRIPTION
**Pull request by hub-cap**
- Fixes Github issues #46 and #45
- Created a middleware for context
- Commented out a bug that was introduced
- Updated context in all services
- Turned a few straggling 201s into 200s

### CODE REVIEW

**Comment by cp16net** [comment]
+1

File: reddwarf/common/wsgi.py [hunk 1]
```diff
@@ -25,11 +25,12 @@
 import webob.dec
 import webob.exc
 
+from reddwarf.common import context as rd_context
 from reddwarf.common import exception
 from reddwarf.common import utils
 from reddwarf.openstack.common import wsgi as openstack_wsgi
 
-
+ContextKey = 'reddwarf.context'
```
Comment (TimSimpsonR) [inline reddwarf/common/wsgi.py]: According to PyLint, as a constant this should be all caps.

**Comment by TimSimpsonR** [comment]
Awesome work! 

P.S. Thanks for fixing my bug.

+1

File: reddwarf/common/wsgi.py (hunk 1 above)
Comment (hub-cap) [inline reddwarf/common/wsgi.py]: Easy enough. I'll fix it right up.

**Pull request merged**